### PR TITLE
Feature/classes crud operations

### PR DESCRIPTION
**Pull request by katherineyevsukov**
## Description

- installed yup and moment as dependencies
- created schema for classes
- removed manual validation middleware and replaced with yup validation
- updated middleware and endpoints related to adding, editing, and deleting classes
- added swagger documentation related to new endpoints
- took out owner authorization and role authentication middleware for now, as refactor will be necessary

## Loom Video

https://www.loom.com/share/03067715994a481881bd06c3be44f305

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: api/classInstances/classInstanceMiddleware.js [hunk 1]
```diff
@@ -25,107 +20,34 @@ const checkClassInstanceExist = async (req, res, next) => {
...
-  const courseType = await getAllCourseTypes()
-    .where({ course_type_id })
-    .first();
-  if (!courseType)
-    next({
-      status: 400,
-      message: `courseType with id ${course_type_id} does not exist`,
+const validateClassObject = async (req, res, next) => {
+  try {
+    const validated = await classSchema.validate(req.body, {
```
Comment (waylonturbes) [inline api/classInstances/classInstanceMiddleware.js]: Great job utilizing `yup` to validate classes. It reduces a lot of code, accomplishes the same goal, is way more readable, and formats a class to be the correct way. Only thing I would change is the name "validated" for the validated class. I would name it "validatedClass" just to make it a bit more clear.

File: api/classInstances/classInstancesRouter.js [hunk 2]
```diff
@@ -225,49 +226,190 @@ router.get('/', authRequired, function (req, res, next) {
...
-          message: 'New Class Instance Added.',
-          schedule: inserted[0],
-        });
+      const inserted = await Classes.addClassInstance(classInstance);
+      res.status(200).json({
+        message: 'New Class Instance Added.',
+        created_class: inserted,
       });
-    } catch (e) {
-      console.error(e);
```
Comment (waylonturbes) [inline api/classInstances/classInstancesRouter.js:251]: Good job getting rid of that log.

**Review by mcmillanj — APPROVED** [review]
great job with schema for classes, and the implementation of  yup for validation and  post put and the delete enpoints. also a great job on the  swagger

**Review by waylonturbes — APPROVED** [review]
Great work adding more `swagger` documentation. I also really like how you are using `yup` to validate classes. Excellent improvements Kat!

File: api/classInstances/classInstanceMiddleware.js (hunk 1 above)
Comment (katherineyevsukov) [reply api/classInstances/classInstanceMiddleware.js]: Great point! I will update it :)

File: api/classInstances/classInstancesRouter.js [hunk 3]
```diff
@@ -225,49 +226,190 @@ router.get('/', authRequired, function (req, res, next) {
...
+ *                  end_time: '12:30:00'
+ *                  start_date: '2022-04-04T04:00:00.000Z'
+ *                  end_date: '2022-04-28T04:00:00.000Z'
+ *                  location: 'Childrens Coding Center'
+ *                  number_of_sessions: 4
+ *     401:
+ *         $ref: '#/components/responses/UnauthorizedError'
+ *     403:
+ *         $ref: '#/components/responses/UnauthorizedError'
+ */
```
Comment (paulstgermain) [inline api/classInstances/classInstancesRouter.js:301]: Nice work adding this Swagger documentation - always nice to have for future teams coming in, so we really appreciate your focus on documentation here!

**Pull request merged**